APIMF-3503: unify example emission behavior, remove LocalElement annotation

arielmirra opened the pull request:
- APIMF-3503: refactor shape canonizer
- APIMF-3503: unify and fix examples propagation in simple inheritance
- chore: refactor amfclient test to amf model assertion test
- APIMF-3503: delete LocalElement annotation

Commit:
APIMF-3503: refactor shape canonizer

Commit:
APIMF-3503: unify and fix examples propagation in simple inheritance

Commit:
chore: refactor amfclient test to amf model assertion test

tomsfernandez commented on amf-shapes/shared/src/main/scala/amf/shapes/internal/domain/resolution/shape_normalization/ShapeCanonizer.scala:
```diff
@@ -112,7 +113,7 @@ sealed case class ShapeCanonizer()(implicit val context: NormalizationContext) e
 
   protected def canonicalInheritance(shape: Shape): Shape = {
-    if (endpointSimpleInheritance(shape)) {
+    if (notDeclaredSimpleInheritance(shape)) {
```
prefer affirmative variable names over negative

tomsfernandez commented on amf-shapes/shared/src/main/scala/amf/shapes/internal/domain/resolution/shape_normalization/ShapeCanonizer.scala:
```diff
@@ -239,19 +235,48 @@ sealed case class ShapeCanonizer()(implicit val context: NormalizationContext) e
+      case _              => None
+    }
+
+    shape match {
```
Probably better to pattern match against a tuple ? (Shape, Option[Shape]) ? Or leave the Seq

Commit:
APIMF-3503: delete LocalElement annotation

- delete fromLocal uses
- now propagate examples in simple inheritance situations

Commit:
chore: update package locks to lockfile 2
